SCClassLibrary/Common/Core/Object.sc

```diff
@@ -217,6 +217,20 @@ Object  {
 	hash { _ObjectHash; ^this.primitiveFailed }
 	identityHash { _ObjectHash; ^this.primitiveFailed }
 
+	// lazy equality: same as == for objects
+	// "composed" for lazy operands (patterns, UGens)
+	|==| { |that|
+		^that.prReverseLazyEquals(this)
+	}
+	|!=| { |that|
+		^not(this |==| that)
+	}
+	// a user might write `something |==| aPattern`
```

Comment:
This comment isn't so instructive, I find, or even necessary (we don't have it for the other dispatched operations either) – but it would be good to explain the reasoning why `reverseComposeBinaryOp` isn't of help.

Reply:
if you feel like it, you could rephrase it: 
```
// we might write `something |==| aPattern`
```
(not perpetuating user-developer separation unintentionally)
but not an issue of course.